omnipaxos/src/ballot_leader_election.rs

```diff
@@ -74,7 +74,9 @@ pub(crate) struct BallotLeaderElection {
     peers: Vec<NodeId>,
     /// The current round of the heartbeat cycle.
     hb_round: u32,
-    /// Vector which holds all the received heartbeats
+    /// Vector which temporarily holds all the received heartbeats from one heartbeat round, including the current node.
+    ballots_temp: Vec<(Ballot, Connectivity)>,
+    /// Vector that holds all the received heartbeats from the previous heartbeat round, including the current node. Represents nodes that are currently alive from the view of the current node.
     ballots: Vec<(Ballot, Connectivity)>,
```

Comment:
I would say we should keep the same field and name for variables as before, and anything we need for the UI we add under the `ui` feature flag with a clear name. So maybe here you need something like
```suggestion
    /// Vector which temporarily holds all the received heartbeats from one heartbeat round, including the current node.
    #[cfg(feature = "ui")]
    prev_round_ballots: Vec<(Ballot, Connectivity)>,
    /// Vector that holds all the received heartbeats from the previous heartbeat round, including the current node. Represents nodes that are currently alive from the view of the current node.
    ballots: Vec<(Ballot, Connectivity)>,
```